src/tests/test_circumcision.py

```diff
@@ -0,0 +1,435 @@
+    pop.data[col.CIRCUMCISION_DATE] = None
+    pop.data[col.VMMC] = False
+    pop.circumcision.init_birth_circumcision_born(pop.data, pop.date)
+
+    # get stats
+    no_circumcised = len(pop.data[pop.data[col.CIRCUMCISED]])
+    male_population = pop.data[(pop.data[col.SEX] == SexType.Male)
+                               & (pop.data[col.AGE] > 0.25)]
+    mean = len(male_population) * pop.circumcision.prob_birth_circ
+    stdev = sqrt(mean * (1 - pop.circumcision.prob_birth_circ))
+    # basic checks
+    general_circumcision_checks(mean, stdev, no_circumcised, pop.data)
+
+    # evolve population
+    while pop.date <= stop_date:
+        # advance ages and birth circumcision
+        pop.data.age += time_step.days / 365
+        pop.circumcision.update_birth_circumcision(pop.data, time_step, pop.date)
+        pop.date += time_step
+
+    # get stats
+    no_circumcised = len(pop.data[pop.data[col.CIRCUMCISED]])
+    male_population = pop.data[(pop.data[col.SEX] == SexType.Male)
+                               & (pop.data[col.AGE] > 0.25)]
+    mean = len(male_population) * pop.circumcision.prob_birth_circ
+    stdev = sqrt(mean * (1 - pop.circumcision.prob_birth_circ))
+    # basic checks
+    general_circumcision_checks(mean, stdev, no_circumcised, pop.data)
+    # check that no circumcised people have undergone VMMC
+    assert len(pop.data[pop.data[col.CIRCUMCISED] & pop.data[col.VMMC]]) == 0
+
+
+def general_circumcision_checks(mean, stdev, no_circumcised, data):
+
+    # no female is marked as circumcised
+    assert len(data[(data[col.SEX] == SexType.Female)
+                    & data[col.CIRCUMCISED]]) == 0
+
+    # no uncircumcised people have circumcision dates
+    assert len(data[~data[col.CIRCUMCISED]
+                    & data[col.CIRCUMCISION_DATE].notnull()]) == 0
+
+    # all circumcised people have circumcision dates
+    assert len(data[data[col.CIRCUMCISED]]) == \
+           len(data[data[col.CIRCUMCISED] & data[col.CIRCUMCISION_DATE].notnull()])
+
+    # check circumcised value is within 3 standard deviations
+    assert mean - 3 * stdev <= no_circumcised <= mean + 3 * stdev
+
+
+def test_vmmc_basic():
+
+    N = 100000
+    start_date = date(2008, 12, 1)
+    stop_date = date(2028, 12, 1)
+    time_step = timedelta(days=90)
+
+    # build artificial population
+    pop = Population(size=N, start_date=start_date)
+    pop.data[col.SEX] = SexType.Male
+    pop.data[col.AGE] = 18
+    pop.data[col.CIRCUMCISED] = False
+    pop.data[col.CIRCUMCISION_DATE] = None
+    pop.data[col.VMMC] = False
+
+    pop.circumcision.circ_inc_rate_year_i = 0
+    pop.circumcision.covid_disrup_affected = 0
+    pop.circumcision.vmmc_disrup_covid = 0
+    pop.circumcision.mc_int = 2008
+    pop.circumcision.year_interv = 2022
+
+    # evolve population
+    pop.data.age += time_step.days / 365
+    pop.circumcision.update_vmmc(pop)
+    # check no VMMC occurs until after mc_int
+    assert len(pop.data[pop.data[col.VMMC]]) == 0
```

Comment:
The date is 1/12/2008, and the introduction of VMMC is 2008, so shouldn't VMMC already be active? 
I one update enough to test this? i.e. do we have enough boys that we expect at least one to get circumcised in this timestep? It might be better to check multiple updates before the VMMC introduction date.